### PR TITLE
refactor(platform): getRawFile and getJsonFile throw instead of null

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

In order to show good error messages, it would be better to throw + handle, instead of return `null` + logging

## Context:

Ref: #9171

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/platform/gitlab/__snapshots__/index.spec.ts.snap [hunk 1]
```diff
@@ -1600,7 +1600,7 @@ Array [
       "user-agent": "https://github.com/renovatebot/renovate",
     },
     "method": "GET",
-    "url": "https://gitlab.com/api/v4/projects/some%2Frepo/repository/files/file.json?ref=HEAD",
+    "url": "https://gitlab.com/api/v4/projects/some%2Frepo/repository/files/dir%2Ffile.json?ref=HEAD",
```
Comment (zharinov) [inline lib/platform/gitlab/__snapshots__/index.spec.ts.snap:1603]: To make `getJsonFile` tests look more uniform, I've intentionally added `dir/` prefix here

File: lib/platform/gitlab/index.spec.ts [hunk 2]
```diff
@@ -1456,17 +1456,17 @@ These updates have all been created already. Click a checkbox below to force a r
...
       const scope = await initRepo();
       scope
-        .get('/api/v4/projects/some%2Frepo/repository/files/file.json?ref=HEAD')
+        .get(
+          '/api/v4/projects/some%2Frepo/repository/files/dir%2Ffile.json?ref=HEAD'
+        )
         .replyWithError('some error');
-      const res = await gitlab.getJsonFile('file.json');
-      expect(res).toBeNull();
+      await expect(gitlab.getJsonFile('dir/file.json')).rejects.toThrow();
```
Comment (zharinov) [inline lib/platform/gitlab/index.spec.ts:1469]: Note: `dir/` prefix was added

File: lib/workers/repository/init/apis.ts [hunk 3]
```diff
@@ -17,13 +17,19 @@ const defaultConfigFile = (config: RenovateConfig): string =>
     ? config.onboardingConfigFileName
     : configFileNames[0];
 
+async function getJsonFile(file: string): Promise<any | null> {
```
Comment (viceice) [inline lib/workers/repository/init/apis.ts:20]:
```suggestion
async function getJsonFile(file: string): Promise<RenovateConfig | null> {
```
should be typed

**Review by viceice — CHANGES_REQUESTED** [review]

**Review by JamieMagee — APPROVED** [review]
LGTM

**Review by viceice — APPROVED** [review]

**Pull request merged**